File: html/data-sources/asp-net-sql-server/asp-net-solution/JPList/Domain/Models/StatusDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace JPList.Domain.Models
{
    /// <summary>
    /// status data transfer object
    /// </summary>
    public class StatusDTO
    {
        /// <summary>
        /// jplist action: paging, filter, sort, etc.
        /// </summary>
        public string action { get; set; }

        /// <summary>
        /// jplist control name
        /// </summary>
        public string name { get; set; }

        /// <summary>
        /// jplist control type: drop-down, textbox, etc.
        /// </summary>
        public string type { get; set; }

        /// <summary>
        /// status related data
        /// </summary>
        public StatusDataDTO data { get; set; }
    }
}

File: source-code/app/dto/status-dto.js
/**
* jPList - jQuery Data Grid Controls ##VERSION## - http://jplist.com 
* Copyright 2014 Miriam Zusin. All rights reserved.
* For non-commercial, personal, or open source projects and applications, you may use jPList for free  http://www.binpress.com/license/read/id/2749/app/2085
* If your project generates any type of income, e.g. sells products, ads, services or just represents a commercial company, you should get a commercial license at http://www.binpress.com
*/
(function(){
	'use strict';		
			
	/**
	* Status
	* @constructor 
	* @param {?string} name
	* @param {?string} action
	* @param {?string} type
	* @param {Object} data
	* @param {boolean} inStorage - is stored in storage (cookies, localstorage, etc.)
	* @param {boolean} inAnimation - is included in animations
	* @param {boolean} isAnimateToTop - is "animate to top" enabled
	* @param {boolean} inDeepLinking - is deep linking enabled for the given control
	*/
	jQuery.fn.jplist.app.dto.StatusDTO = function(name, action, type, data, inStorage, inAnimation, isAnimateToTop, inDeepLinking){
			
		this.action = action; //filter, sort, paging			
		this.name = name;
		this.type = type; //drop-down, pagination, textbox
		this.data = data; //related data
		this.inStorage = inStorage;
		this.inAnimation = inAnimation;
		this.isAnimateToTop = isAnimateToTop;
		this.inDeepLinking = inDeepLinking;
	};
})();

